convert independent tools to single binary with subcommands (#10)
kyma-project/eventing-tools: docs/validationtest/README.md

# Validationtest

> This set of tools is used to publish continues legacy and cloudevents the Eventing publisher proxy Kyma. In addition to the publisher

Reviewer: ```suggestion
This set of tools continuously publishes legacy and CloudEvents the Eventing publisher proxy in Kyma. In addition to the publisher, a set of subscribers is created with corresponding subscriptions.
```

> a set of subscribers is created with corresponding subscriptions.

Reviewer: ```suggestion
```

## Application to event type mapping

> Currently, it is hardcoded which application sends which event type as follows:

Reviewer: ```suggestion
Currently, it is hardcoded which application sends which event type:
```

| Application   | Event type                                       |
|---------------|--------------------------------------------------|
| commerce      | order.created                                    |
| appname       | DocuSing_BO.Account_DocuSign.Updated             |
| no-app        | New.Some-Other.Order-äöüÄÖÜβ.Final.C-r-e-a-t-e-d |

## Subscription

> 

Reviewer: Can we add an introductory sentence to this list?

Author: i tried ;)

Reviewer: Excellent, thank you!

1. **Three** microservice subscribers.

> 2. **Three** Kyma function subscribers.

Reviewer: Can't comment on line 20, but I suggest the following punctuation:

`3. **Six** Kyma subscriptions; each of them has the following hardcoded filters:`

| eventing-test   | function-subscription-1 | function-1         |
| eventing-test   | function-subscription-2 | function-2         |

## Usage

> ### Deploy

Reviewer: ```suggestion
### Deploy Validationtest 
```

1. Connect to a Kubernetes cluster.
2. Execute:
   ```bash
   make validationtest-deploy
   ```

> ### Stop

Reviewer: ```suggestion
### Stop Validationtest
```

> This is useful in case we want to stop sending events to the Eventing publisher proxy, but preserve the Eventing

Reviewer: ```suggestion
You can stop sending events to the Eventing publisher proxy while preserving the Eventing infrastructure inside the `eventing-test` Namespace with the following command:
```

> infrastructure inside the `eventing-test` namespace.

Reviewer: ```suggestion
```

1. Connect to a Kubernetes cluster.
2. Execute:
   ```bash
   make validationtest-stop
   ```

   > Note: To continue sending events, deploy the publisher again.

### Logs

> #### Publisher Logs

Reviewer: ```suggestion
#### View Publisher Logs
```
I'm guessing these steps are to view the logs?

1. Connect to a Kubernetes cluster.
2. Publisher logs:
   ```bash
   stern -n eventing-test publisher -c publisher --since 1s
   ```

> #### Subscriber Logs

Reviewer: ```suggestion
#### View Subscriber Logs
```

> 1. Connect to a Kubernetes cluster.

Reviewer: In line 79, we could write "Remove Validationtest" to distinguish it from "Stop Validationtest".

however, if "Cleanup" is always the last instruction in your docs, ignore my suggestion :)
